Migrate to Capacitor 4

Nevvulo opened the pull request:
Updates the plugin to support Capacitor 4.
I've confirmed this is working for Android and iOS, tested in the simulator and through `npm run verify`

## Capacitor 3 vs Capacitor 4

- Version 0.1.3 (and lower) will target the latest Capacitor version
- Version 1.0.0 will target Capacitor 4

## Breaking Changes
* Plugin must be imported like `import { Segment } from '@joinflux/capacitor-segment'` (instead of `CapacitorSegment`)
* Version 1.x.x will solely support Capacitor 4
* No functionality has changed

anonimitoraf commented on android/src/main/AndroidManifest.xml:
```diff
@@ -1,3 +1,3 @@
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
-    package="com.joinflux.flux.capacitorsegment">
```
Just wondering why the package name has changed?

anonimitoraf commented on android/src/main/java/com/joinflux/flux/segment/SegmentPlugin.java:
```diff
@@ -26,24 +26,27 @@ public void initialize(PluginCall call) {
         }
 
         Context context = this.getContext();
-        Builder builder = new Analytics.Builder(context, key);
```
Why was this code moved back into the UI thread? (i.e. this [commit](https://github.com/joinflux/capacitor-segment/commit/62555b325a770bc77d70e3c4ec7f98797b8434b0) seems to have been reverted)?

Nevvulo replied:
Thanks for picking this up, this was not meant to be re-added, will fix this

anonimitoraf reviewed (commented):
Nice job on this :) 

I just left a couple of questions

anonimitoraf commented on android/gradlew:
```diff
@@ -1,7 +1,7 @@
-#!/usr/bin/env sh
```
I should've asked this earlier: does gradlew get auto-generated?
I wonder why the shebang changed sh's path. The latter is a bit less robust AFAIK (some systems don't have sh under `/bin`)

Nevvulo replied:
It does get automatically generated when creating a new plugin, which is what I've done for migrating to Capacitor 4. I'm not sure why this has changed but it hasn't caused any issues for me personally while building or testing the updated plugins in-app

anonimitoraf replied:
Ah ok I see. And yeah, most (POSIX) systems would have /bin/sh but it's encouraged best practice to use /usr/bin/env

gugahoi commented on README.md:
```diff
@@ -23,15 +23,20 @@ and
 
+### Capacitor 3 vs Capacitor 4
+
+- Version 0.1.3 will target the latest Capacitor version
```
Should we specify that we recommend Capacitor 3 for this version?

Nevvulo replied:
Sure, I'll update this

gugahoi commented on android/build.gradle:
```diff
@@ -36,22 +36,21 @@ android {
     compileOptions {
-        sourceCompatibility JavaVersion.VERSION_1_8
-        targetCompatibility JavaVersion.VERSION_1_8
+        sourceCompatibility JavaVersion.VERSION_11
+        targetCompatibility JavaVersion.VERSION_11
```
Is this correct? I believe 1 of them is OpenJDK and the other is Oracle JDK.

Nevvulo replied:
This is directly from a fresh Capacitor 4 plugin template from `npx @capacitor/create-plugin`, see [here](https://github.com/ionic-team/create-capacitor-plugin/blob/03027bf603626ebfdba084cb9d9a1a359d008554/assets/plugin-template/android/build.gradle#L39-L40). I haven't had any issues when testing however I did need to update my Android Studio to use the updated version mentioned 
![image](https://user-images.githubusercontent.com/26913683/191170858-9ebfec2f-966a-45dd-a6f1-6d4a7323207d.png)

gugahoi replied:
I see, no worries.

gugahoi commented on android/src/main/java/com/joinflux/flux/segment/SegmentPlugin.java:
```diff
@@ -26,24 +26,27 @@ public void initialize(PluginCall call) {
+                        implementation.analytics = builder.build();
+                    }
+                }
+            );
```
Assuming this was copied from Segment documentation? If not, do we have good reason for this change?

Nevvulo replied:
This was a mistake, I've reverted this back to the way we had it before (without the need for UI thread). It should be fixed in [73c7b79](https://github.com/joinflux/capacitor-segment/pull/14/commits/73c7b79a7bf6deeac53eba4a7f6691b1247d4e98)

gugahoi reviewed (commented):
Looks mostly good, I have a couple of questions.

Nevvulo commented:
Planning to merge this shortly as I believe I've addressed concerns. I've tested the plugin on iOS + Android (simulator + real device) and everything worked as expected
